api/src/services/script-runner.service.ts

```diff
@@ -28,34 +31,541 @@ export class ScriptRunnerService {
+                      id: doorwayAmiChart.id,
+                    },
+                  }
+                : undefined,
+              unitTypes: unitType
+                ? {
+                    connect: {
+                      id: unitType.id,
+                    },
+                  }
+                : undefined,
+              unitAccessibilityPriorityTypes: priorityType
+                ? { connect: { id: priorityType.id } }
+                : undefined,
+              unitRentTypes: rentType
+                ? { connect: { id: rentType.id } }
+                : undefined,
+              amiPercentage: unit['ami_percentage'],
+              annualIncomeMin: unit['annual_income_min'],
+              annualIncomeMax: unit['annual_income_max'],
+              floor: unit['floor'],
+              monthlyIncomeMin: unit['monthly_income_min'],
+              maxOccupancy: unit['max_occupancy'],
+              minOccupancy: unit['min_occupancy'],
+              monthlyRent: unit['monthly_rent'],
+              numBathrooms: unit['num_bathrooms'],
+              numBedrooms: unit['num_bedrooms'],
+              number: unit['number'],
+              sqFeet: unit['sq_feet'],
+              monthlyRentAsPercentOfIncome:
+                unit['monthly_rent_as_percent_of_income'],
+              bmrProgramChart: unit['bmr_program_chart'],
+              status: unit['status'],
+            },
+          });
+        });
+
+        // Migrate the listing to multiselect question mapping
+        const listingMultiselectQuestions: {
+          ordinal: number;
+          listing_id: string;
+          multiselect_question_id: string;
+        }[] = await client.$queryRawUnsafe(
+          `SELECT * FROM listing_multiselect_questions WHERE listing_id = '${listing['id']}'`,
+        );
+        if (listingMultiselectQuestions?.length) {
+          console.log(
+            `migrating ${listingMultiselectQuestions.length} listing multiselect questions`,
+          );
+          listingMultiselectQuestions.forEach(async (lmq) => {
+            try {
+              await this.prisma.listingMultiselectQuestions.create({
+                data: {
+                  ordinal: lmq.ordinal,
+                  listingId: createdListing.id,
+                  multiselectQuestionId: lmq.multiselect_question_id,
+                },
+              });
+            } catch (e) {
+              // Log the failed ones so we can manually add them later if need be
+              console.log(
+                `unable to migrate listing multiselect question ${lmq.multiselect_question_id} to listing ${createdListing.id}`,
+              );
+            }
+          });
+        }
+
+        // Migrate all events that don't have a file associated to it
+        const listingEvents: any[] = await client.$queryRawUnsafe(
+          `SELECT * FROM listing_events WHERE listing_id = '${listing['id']}' AND file_id IS NULL`,
+        );
+        if (listingEvents?.length) {
```

Comment:
And maybe missing from the tests as well

Reply:
new test added!